### PR TITLE
Add scripts to update examples to cypress@latest

### PR DESCRIPTION
**Pull request by MikeMcC399**
This PR implements the suggestion from https://github.com/cypress-io/github-action/issues/767 "Automated examples update to latest Cypress version" by providing a new npm script command:

```bash
npm run update:cypress
```

to update Cypress examples to the latest version of Cypress. (The v9 examples are untouched and left at version 9.7.0.)

The process is documented in [docs/MAINTENANCE.md](https://github.com/MikeMcC399/github-action/blob/build/update-cypress-latest/docs/MAINTENANCE.md).

1. `update-cypress-latest.sh` is the main script
2. `update-cypress-latest-npm.sh` updates examples using `npm`
3. `update-cypress-latest-other.sh` updates examples using `yarn` and `pnpm`

The `npm run check:markdown` script is extended to check also the new `docs/` directory.

The previous script [scripts/npm-install-examples.sh](https://github.com/cypress-io/github-action/blob/master/scripts/npm-install-examples.sh) is replaced.

### CODE REVIEW

**Comment by MikeMcC399** [comment]
Tested locally on Ubuntu 22.04 and Windows 11 Pro.
Leaving in Draft state until PR backlog smaller.

**Comment by MikeMcC399** [comment]
@jaffrepaul 

I successfully ran this to update to Cypress 12.5.1, so from my point of view this PR is ready to merge.

It's also non-critical in the sense that it is purely a maintenance tool. It is not called by the action itself or by the examples.

I haven't submitted a PR with the actual update to Cypress 12.5.1 waiting in branch [update/examples-to-12-5-1](https://github.com/MikeMcC399/github-action/tree/update/examples-to-12-5-1). Perhaps 12.6. is around the corner and it would be better to wait?

**Comment by MikeMcC399** [comment]
This is now modified to save exact versions of Cypress i.e. `12.7.0` not `^12.7.0` also if the scripts are run directly, for instance
`./scripts/update-cypress-latest-npm.sh` instead of being run using `npm run cypress:update`.

In the latter case they already took the setting `save-exact=true` from [.npmrc](https://github.com/cypress-io/github-action/blob/master/.npmrc).

**Review by mjhenkes — APPROVED** [review]

File: docs/MAINTENANCE.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+This document describes updating the [examples](../examples) in this repository to use the latest Cypress version.
+
+## Examples
+
+There are two groups of examples which are used to test and demonstrate the use of [cypress-io/github-action](https://github.com/cypress-io/github-action):
+
+1. The [examples](../examples) directory contains examples of the use of Cypress (Current) [Configuration](https://docs.cypress.io/guides/references/configuration) which applies to Cypress 10 and later.
+
+2. The [examples/v9](../examples/v9) directory contains examples which are set up to use Cypress 9.7.0 which is the last version using [Legacy Configuration](https://docs.cypress.io/guides/references/legacy-configuration) which applies to Cypress 9 and below.
```
Comment (AtofStryker) [inline docs/MAINTENANCE.md:11]:
```suggestion
2. The [examples/v9](../examples/v9) directory contains examples which are set up to use Cypress `9.7.0` which is the last version using [Legacy Configuration](https://docs.cypress.io/guides/references/legacy-configuration) which applies to Cypress 9 and below.
```

**Review by AtofStryker — APPROVED** [review]

File: docs/MAINTENANCE.md (hunk 1 above)
Comment (MikeMcC399) [reply docs/MAINTENANCE.md]: @AtofStryker 

Unfortunately you were too late with your suggestion and @mjhenkes had already merged. I don't think it is worth a new PR to make the change, but I will bear it in mind if the file needs editing later for some other reason.

Comment (mjhenkes) [reply docs/MAINTENANCE.md]: I applied the suggestion here: https://github.com/cypress-io/github-action/pull/779/commits/9ca965cadc67714f1f7812a449ea48bcf1be5795

Before i merged!

**Pull request merged**